pkg/operator/ceph/nfs/nfs.go

```diff
@@ -289,22 +289,3 @@ func validateGanesha(context *clusterd.Context, clusterInfo *cephclient.ClusterI
 
 	return nil
 }
-
-// create and enable default RADOS pool
-func (r *ReconcileCephNFS) createDefaultNFSRADOSPool(n *cephv1.CephNFS) error {
```

Comment:
@BlaineEXE Instead of creating the default .nfs pool, how about if we require them to define the CephBlockPool?
1. I added the example yaml to the nfs.yaml example. Though that keeps the default example simple, there may be a better place. If another nfs server is added, they shouldn't create the pool again.
2. Do we need to add a wait for the pool in case it doesn't exist, or will it naturally fail the reconcile and retry?

Reply:
This is a good question. I think it is a better experience for users if we don't require them to create the `CephBlockPool` for the shared config. For most users, having that pool created with the default settings should be sufficient I'd think.

I wonder if it makes sense to have the special `CephBlockPool` that controls the `.nfs` pool wait until the pool exists to change the config. I.e., wait for at least one CephNFS reconcile to run. Otherwise, from the CephBlockPool reconciler, we will have to make sure the `nfs` application gets set on the pool.

Reply:
> I wonder if it makes sense to have the special `CephBlockPool` that controls the `.nfs` pool wait until the pool exists to change the config. I.e., wait for at least one CephNFS reconcile to run. Otherwise, from the CephBlockPool reconciler, we will have to make sure the `nfs` application gets set on the pool.

 I'm thinking wait in the nfs controller for the .nfs pool to exist, then the nfs controller can apply the nfs application. Otherwise, there are some pool settings that might not be applied properly. Not all the settings on the pool CR can be modified after pool creation.

Reply:
> I'm thinking wait in the nfs controller for the .nfs pool to exist, then the nfs controller can apply the nfs application. Otherwise, there are some pool settings that might not be applied properly. Not all the settings on the pool CR can be modified after pool creation.

Hmm... okay. That sounds okay. I think the only thing I notice that is missing then is adding the pool to `nfs-test.yaml`.